app/model/block.py

```diff
@@ -10,11 +10,73 @@ def __init__(self):
         self.container = None
         self.questionnaire = None
         self.validation = None
-        self.questionniare = None
+        self.questionnaire = None
         self.templatable_properties = []
         self.display = Display()
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
             section.container = self
+
+    def to_json(self):
```

Comment:
Could still be in scope... I'm not sure.  Currently the to_json() method works to output json in the format expected by the parser, so the round trip can be done through that as seen in the unit-tests.

It's not quite as clean as having a from_json() method, but while so much of this is up for grabs, I'm not sure how much value it adds at this stage to the implement that method.

Any thoughts from @dhilton, @LJBabbage ?